src/ECCX08.cpp

```diff
@@ -548,6 +548,54 @@ int ECCX08Class::nonce(const byte data[])
   return challenge(data);
 }
 
+long ECCX08Class::incrementCounter(int keyId)
+{
+  uint32_t counter;  // the counter can go up to 2,097,151
+
+  if (!wakeup()) {
+    return -1;
+  }
+
+  if (!sendCommand(0x24, 1, keyId)) {
+    return -1;
+  }
+
+  delay(20);
+
+  if (!receiveResponse(&counter, sizeof(counter))) {
+    return -1;
+  }
+
+  delay(1);
+  idle();
+
+  return counter;
+}
```

Comment:
To keep changes aligned to the style of the library i would split this function in two:
* one basic function returning 0 or 1 and taking the counter value as argument;
* another one that wraps it and returns the counter value

 Also, reading the docs it looks to me that keyId can be only 0 or 1 so i would add a check to keyId parameter.

```suggestion
int ECCX08Class::incrementCounter(int keyId, long& counter)
{
  if (keyId < 0 || keyId > 1) {
    return 0;
  }
  
  if (!wakeup()) {
    return 0;
  }

  if (!sendCommand(0x24, 1, keyId)) {
    return 0;
  }

  delay(20);

  if (!receiveResponse(&counter, sizeof(counter))) {
    return 0;
  }

  delay(1);
  idle();

  return 1;
}

long ECCX08Class::incrementCounter(int keyId)
{
  long counter;  // the counter can go up to 2,097,151
  
  if(!incrementCounter(keyId, counter)) {
    return -1;
  }

  return counter;
}
```